fuel-merkle/src/sparse/merkle_tree.rs

```diff
@@ -33,6 +33,57 @@ impl<StorageError> From<StorageError> for MerkleTreeError<StorageError> {
     }
 }
 
+/// The safe Merkle tree storage key prevents Merkle tree structure manipulations.
+/// The type contains only one constructor that hashes the storage key.
+#[derive(Debug, Clone, Copy)]
```

Comment:
should we also include ops like `PartialEq` or `Display`? [derive_more](https://jeltef.github.io/derive_more/derive_more/) may be helpful for newtype ergonomics here.